apis/v1beta1/validation/gateway_test.go

```diff
@@ -136,14 +136,37 @@ func TestValidateGateway(t *testing.T) {
 		},
 		"names are not unique within the Gateway": {
 			mutate: func(gw *gatewayv1b1.Gateway) {
+				hostnameFoo := gatewayv1b1.Hostname("foo.com")
+				hostnameBar := gatewayv1b1.Hostname("bar.com")
 				gw.Spec.Listeners[0].Name = "foo"
-				gw.Spec.Listeners = append(gw.Spec.Listeners, gatewayv1b1.Listener{
-					Name: "foo",
-				},
+				gw.Spec.Listeners[0].Hostname = &hostnameFoo
+				gw.Spec.Listeners = append(gw.Spec.Listeners,
+					gatewayv1b1.Listener{
+						Name:     "foo",
+						Hostname: &hostnameBar,
+					},
 				)
 			},
 			expectErrsOnFields: []string{"spec.listeners[1].name"},
 		},
+		"combination of port, protocol, and name are not unique for each listener": {
+			mutate: func(gw *gatewayv1b1.Gateway) {
+				hostnameFoo := gatewayv1b1.Hostname("foo.com")
+				gw.Spec.Listeners[0].Name = "foo"
+				gw.Spec.Listeners[0].Hostname = &hostnameFoo
```

Comment:
Since hostname is optional can you add one more test case that shows that the same setup fails when both listeners have separate hostnames?

Reply:
Similarly, it would be helpful to have additional test cases where everything was the same except for one of the fields (port, protocol, or hostname) to ensure those are all still valid.

Reply:
> Since hostname is optional can you add one more test case that shows that the same setup fails when both listeners have separate hostnames?

Do you mean to test the combination of port and protocol are unique when hostnames not set? If yes, I've add one: https://github.com/kubernetes-sigs/gateway-api/pull/1457/commits/d75ab5690eab043ce5e3dd3e4f495ec5e398b2b0#diff-e3c0a139e14037364db162e955588e8153e795f4cbe84b6fce32149d62ca2a73R170-R184



> Similarly, it would be helpful to have additional test cases where everything was the same except for one of the fields (port, protocol, or hostname) to ensure those are all still valid.

Updated, thank you!